### PR TITLE
test: List widget UI test cases

### PR DESCRIPTION
**Pull request by arslanhaiderbuttar**
## Description

Automated List Widget UI Test Cases:

- Change the List Widget Name
- Add or Modify items
- Change the Background Color
- Change the Item Background Color
- Enable Disable Visible
- Enable Disable Visible - JS
- Copy Widget
- Delete Widget
- Change the list and item background color using JS

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

**Comment by NandanAnantharamu** [comment]
@arslanhaiderbuttar Copy, delete, enable & disable tests are already present for different widgets, let's wait for feedback, and then we can actually remove them.

cc: @yatinappsmith

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe("Button Widget Functionality", function() {
+  before(() => {
+    cy.addDsl(dsl);
+  });
+
+  beforeEach(() => {
+    cy.openPropertyPane("buttonwidget");
+  });
+
+  it("Button-Style Validation", function() {
```
Comment (NandanAnantharamu) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js:13]: We have a button style check already in button spec, this test is redundant.

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js [hunk 2]
```diff
@@ -10,14 +10,58 @@ describe("Container Widget Functionality", function() {
...
+    cy.get(widgetsPage.listWidget).should("not.exist");
+    cy.get(publishPage.backToEditor).click({ force: true });
+  });
+
+  it("Toggle JS - List-Check Visible field Validation", function() {
+    // Open Property pane
+    cy.SearchEntityandOpen("List1");
+    //Check the disabled checkbox using JS and Validate
+    cy.EditWidgetPropertiesUsingJS(widgetsPage.inputToggleVisible, "true");
+    cy.PublishtheApp();
```
Comment (NandanAnantharamu) [inline app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:49]: Is it necessary to publish the app for each check, cant we do all checks by one publish app?

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js [hunk 3]
```diff
@@ -10,14 +10,58 @@ describe("Container Widget Functionality", function() {
     cy.addDsl(dsl);
   });
 
+  it("List-Unckeck Visible field Validation", function() {
+    // Open Property pane
+    cy.SearchEntityandOpen("List1");
+    //Uncheck the disabled checkbox and validate
+    cy.UncheckWidgetProperties(commonlocators.visibleCheckbox);
+    cy.PublishtheApp();
+    cy.get(widgetsPage.listWidget).should("not.exist");
```
Comment (NandanAnantharamu) [inline app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:19]: check and uncheck are there i guess, can you check once?

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js (hunk 2 above)
Comment (arslanhaiderbuttar) [reply app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:49]: How can we check visibility and invisibility by publishing only once?

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js (hunk 1 above)
Comment (arslanhaiderbuttar) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js:13]: Yes. I know. I am checking it for every widget.

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js (hunk 3 above)
Comment (arslanhaiderbuttar) [reply app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:19]: Not in the List_spec.js

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js:13]: @yatinappsmith we discussed this in today's call, should we be adding for each widget? lets do it if you want

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js (hunk 2 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:49]: fine if we cant do it then its fine, please go ahead

File: app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js (hunk 3 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Smoke_TestSuite/LayoutWidgets/List_spec.js:19]: ok fine

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js (hunk 1 above)
Comment (yatinappsmith) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/List_spec.js:13]: Ideally we should not add a same test case for each widget. Ex copy paste delete. But for bind we should test for each widget

**Review by NandanAnantharamu — DISMISSED** [review]
LGTM

**Review by yatinappsmith — DISMISSED** [review]

**Review by yatinappsmith — DISMISSED** [review]
LGTM

**Review by yatinappsmith — APPROVED** [review]

**Comment by yatinappsmith** [comment]
LGTM

**Pull request merged**